### ISSUE TITLE
yolov11 mask segmenation to rice field

### ISSUE THREAD

**Issue by dimasadef72** [ultralytics/ultralytics#22171]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I want to perform rice field segmentation detection.

For training, I took several rice field plots, then performed segmentation in Roboflow and training in Colab.

The inference input is a large image of several rice fields in a subdistrict area, where there are several small rice fields in that subdistrict. Are there any recommendations on how to make YOLO detect small rice fields in the large map image?

Translated with [DeepL.com](https://www.deepl.com/?utm_campaign=product&utm_source=web_translator&utm_medium=web&utm_content=copy_free_translation) (free version)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @dimasadef72, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will follow up to assist you soon. In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (e.g., YOLO11 segmentation on large imagery with small fields), please share as much detail as possible to help us guide you:
-Your exact training/inference commands and YOLO11 model variant used
-Image sizes/resolution for training and inference
-A few annotated sample images and corresponding predictions
-Training logs (epochs, metrics) and environment details (CPU/GPU, CUDA, PyTorch)
-Any preprocessing steps applied to large maps and any strategies you’ve tried

Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Great question—use tiled/sliced inference so YOLO11 can “see” small plots inside very large orthoimages; SAHI integrates directly with Ultralytics and stitches results back together, see the guide at [Using YOLO11 with SAHI for Sliced Inference](https://docs.ultralytics.com/guides/sahi-tiled-inference/). 

```python
from ultralytics import YOLO
from sahi import AutoDetectionModel
from sahi.predict import get_sliced_prediction

# Load your YOLO11 instance-seg model
model_path = "yolo11n-seg.pt"  # choose n/s/m/l/x-seg as needed
detection_model = AutoDetectionModel.from_pretrained(
    model_type="ultralytics", model_path=model_path, confidence_threshold=0.25, device="cuda:0"
)

# Run sliced inference on a large map
result = get_sliced_prediction(
    "subdistrict.png", detection_model,
    slice_height=1024, slice_width=1024, overlap_height_ratio=0.2, overlap_width_ratio=0.2
)
result.export_visuals(export_dir="runs/seg-sahi")
```

As a quick baseline you can also try higher-resolution inference: `YOLO("yolo11n-seg.pt").predict("subdistrict.png", imgsz=2048)`. For training tips and mask outputs, the task reference is here: [Segmentation task docs](https://docs.ultralytics.com/tasks/segment/).

**Comment by dimasadef72** [comment]
What is the difference between sahi and regular tiled?

**Comment by glenn-jocher** [comment]
SAHI is tiled inference done for you: it slices with overlap and automatically remaps, merges, and de‑duplicates cross‑tile predictions (boxes and masks); “regular tiled” means you crop tiles yourself and must handle overlap, boundary cuts, coordinate offsets, and cross‑tile NMS/mask stitching manually, which is easier to get wrong at edges.
